Scala/sRAPL/sRAPL.scala

```diff
@@ -0,0 +1,199 @@
+// using scala 3.1
+// using lib jRAPL::jRAPL:1.0
+
+package sRAPL
+
+import jRAPL._
+import java.io._
+
+@main def test(): Unit = {
+given config: Settings = Settings(warmupConfig = WarmupSettings(measureWarmup = true))
+ measureEnergyConsumption("test"){ () =>
+   (0 until 10).foreach{_ =>
+   util.Random.alphanumeric.take(1000000).count(_ == 'A')
+   }
+ } 
+}
+
+def measureEnergyConsumption(
+    label: String
+)(fn: () => Unit)(using config: Settings): Unit = {
+  withFreshSyncMonitor {
+    withConfigPrintStream {
+      val warmupConfig = config.warmupConfig
+      for
+        i <- 0 until warmupConfig.iterations
+        // input = warmupConfig.input.fold(
+        //   throw RuntimeException("Warmup input not defined")
+        // )(_.inputStream())
+      do
+        println(s"$label warmup - iteration $i")
+        if warmupConfig.measureWarmup then
+          measureEnergyConsumptionSample(s"$label-warmup_$i")(fn())
+        else fn()
+      end for
+
+      for i <- 0 until 10
+      do  measureEnergyConsumptionSample(label)(fn())
+    }
+  }
+}
+
+private def measureEnergyConsumptionSample[T](
+    label: String
+)(
+    fn: => T
+)(using output: PrintStream)(using Settings, SyncEnergyMonitor): T = {
+  val (sample, result) = executeAndMeasureEnergyConsumption(fn)
+  output.println(sample.toCSV(label))
+  result
+}
+
+private def executeAndMeasureEnergyConsumption[T](
+    fn: => T
+)(using monitor: SyncEnergyMonitor): (EnergyDiff, T) = {
+  val before = monitor.getSample()
+  val result = fn
+  val after = monitor.getSample()
+  EnergyDiff.between(before, after) -> result
```

Comment:
So, all of that delegates to jRAPL, right? It seems that there is very little we can do to investigate why we got weird results…